README.adoc

```diff
@@ -0,0 +1,129 @@
+= The Agriculture Core
+
+_The Agriculture Core_ is a NodeJS server exposing a robust REST API for various client applications to connect to.
+
+== Configuration
+
+The agriculture-core pulls its configuration from a `.env` file at the root of the project.
+These configuration variables are loaded into `process.env.<VAR_NAME>` and are accessible accross the application.
+To get started, this configuration file must be created with the following format:
+
+[source,properties]
+----
+# Application Configuration
+PORT=
+
+# Database Environment
+DB_HOST=
+DB_NAME=
+DB_USER=
+DB_PASS=
+
+# Authentication
+JWT_SECRET=
+JWT_ISSUER=
+JWT_AUDIENCE=
+JWT_EXPIRES=
+
+# Logging
+LOG_LEVEL=
+----
+
+`LOG_LEVEL` can be configured from levels 0-5:
+
+ * error: 0
+ * warn: 1
+ * info: 2
+ * verbose: 3
+ * debug: 4
+ * silly: 5
+
+== Development
+
+=== Dependencies
+
+* link:https://nodejs.org/en/[NodeJS] is the framework in use.
+* MySQL server our some other valid host indicated by the `.env` file.
+* link:https://yarnpkg.com/lang/en/docs/install/[yarn] handles all other dependencies.
+
+To install all other dependencies with use the command `yarn`.
+
+=== Building
+
+IMPORTANT: `npm start` runs a file watcher already, no other build steps are required
+
+To build and run for production run:
+
+	npm run start:prod
+
+To just build for production run:
+
+	npm run build
+
+An equivalent explicit command is also available:
+
+	npm run build:prod
+
+To build and run for development:
+
+	npm run build:dev
+
+To just watch the changes and build them when they change (for development), use:
+
+	npm run watch
+
+`npm start` already performs this so it's not necessary to run both.
+
+==== Testing
+
+=== Naming Conventions
+
+Creating a new test file is easy!
+You can add your unit tests in `<ROOT>/test/unit` and integration tests in `<ROOT>/test/integration`.
+Just make sure your files end with `.test.ts` so they will be picked up in the webpack build.
+
+Unit tests go in `agriculture-core/test/unit/`.
+Integration tests go in `agriculture-core/test/integration`.
+To ensure the files are discovered by the webpack build, test files must have the suffix `.test.ts`.
```

Comment:
Did you intend to get rid of my lines above? I missed that these are repeated before

Reply:
Yup, did intend to do that.